# HG changeset patch
# User Rik <rik@octave.org>
# Date 1427825926 25200
#      Tue Mar 31 11:18:46 2015 -0700
# Node ID 91e2212b060d3c64d9cd8bb26f7820729517b461
# Parent  abf348f41759a47c6b2c829281ba0f9870fbfa04
Fix FLTK SelectionType for Middle Button and double clicks (bug #44669).

* __init_fltk__.cc (handle): Check event for FL_MIDDLE_BUTTON and return
"extend" if found.  Use Fl::event_clicks to check for double click before
any other button processing and return "open" if found.

diff --git a/libinterp/dldfcn/__init_fltk__.cc b/libinterp/dldfcn/__init_fltk__.cc
--- a/libinterp/dldfcn/__init_fltk__.cc
+++ b/libinterp/dldfcn/__init_fltk__.cc
@@ -1449,25 +1449,26 @@ private:
               return 1;
 
             case FL_PUSH:
               pos_x = Fl::event_x ();
               pos_y = Fl::event_y () - menu_dy ();
 
               set_currentpoint (pos_x, pos_y);
 
-              if (Fl::event_button () == FL_LEFT_MOUSE
-                  && Fl::event_shift ())
+              if (Fl::event_clicks ())
+                fp.set_selectiontype ("open");
+              else if (Fl::event_button () == FL_MIDDLE_MOUSE
+                       || (Fl::event_button () == FL_LEFT_MOUSE
+                           && Fl::event_shift ()))
                 fp.set_selectiontype ("extend");
-              else if ((Fl::event_button () == FL_LEFT_MOUSE
-                        && Fl::event_ctrl ())
-                       || Fl::event_button () == FL_RIGHT_MOUSE)
+              else if (Fl::event_button () == FL_RIGHT_MOUSE
+                       || (Fl::event_button () == FL_LEFT_MOUSE
+                           && Fl::event_ctrl ()))
                 fp.set_selectiontype ("alt");
-              else if (Fl::event_clicks ())
-                fp.set_selectiontype ("open");
               else
                 fp.set_selectiontype ("normal");
 
               gh = pixel2axes_or_ca (pos_x, pos_y);
 
               if (gh.ok ())
                 {
                   ax_obj = gh_manager::get_object (gh);
